api/src/certification/scoring/domain/models/CapacitySimulator.js

```diff
@@ -31,3 +48,24 @@ export class CapacitySimulator {
     });
   }
 }
+
+export function findIntervalIndex({ score, weightsAndCoefficients, scoringIntervalsLength }) {
+  const MAX_PIX_SCORE = config.v3Certification.scoring.maximumReachableScore;
+  const LAST_INTERVAL_INDEX = scoringIntervalsLength - 1;
+
+  if (score >= MAX_PIX_SCORE) {
+    return LAST_INTERVAL_INDEX;
+  }
+
+  const weights = weightsAndCoefficients.map(({ weight }) => weight);
+
+  let cumulativeSumOfWeights = weights[0];
+
+  for (let index = 0; index < weights.length; index++) {
+    if (score < cumulativeSumOfWeights) {
+      return index;
+    } else {
+      cumulativeSumOfWeights += weights[index + 1];
+    }
+  }
+}
```

Comment:
Je trouve  la lecture de la fonction un peu difficile, notamment lié à beaucoup de conditions d'arrêts.

Je propose cette version ci qui semble passer les test 

```suggestion
export function findIntervalIndex({ score, weightsAndCoefficients, scoringIntervalsLength }) {
  const isPixScoreOfAnHigherInterval = (score, nextIntervalMinimumScore) => score >= nextIntervalMinimumScore;
  const hasANextInterval = (currentInterval) => currentInterval < scoringIntervalsLength - 1;

  const weights = weightsAndCoefficients.map(({ weight }) => weight);
  let cumulativeSumOfWeights = weights[0];
  let currentScoringInterval = 0;

  while(isPixScoreOfAnHigherInterval(score, cumulativeSumOfWeights) && hasANextInterval(currentScoringInterval)){
    currentScoringInterval++;
    cumulativeSumOfWeights += weights[currentScoringInterval];
  }

  return currentScoringInterval;
}
```

Reply:
A noter que, dans les deux versions du code, je pense que la ligne `const weights = weightsAndCoefficients.map(({ weight }) => weight);` ne devrait pas faire partie de cette fonction.